### PR TITLE
Fix reset function

### PR DESCRIPTION
**Pull request by bs-github**
This change implements 3 reset stages.

0. Resetting network configuration and disabling web interface login
1. Removing configuration but keeping charge log.
2. Format data partition. (This also removes tracked charges and the username file)

The reset is initiated by pressing button SW3 during boot.  
The stage is selected by the duration of holding the button down.

If you **press and hold SW3 on boot**, the **green LED** will be on for the **first 10 seconds**. If you release the button within this time, **nothing** will be reset and the boot will continue normally. If the button is **still pressed**, the **green LED** will start **flashing** and if SW3 is released within the **next 5 seconds** **_stage 0_ will be initiated**. The **next 5 seconds** both, the **green and the red LED will be flashing** and if released in this period, **_stage 1_ will be entered**. If still pressed, **only the red LED will flash for another 5 seconds** and if released now, **_stage 2_ will be executed**.   
If SW3 is still pressed after this, finally a normal boot will happen without resetting anything.

### CODE REVIEW

**Comment by bs-github** [comment]
This reset logic is directly taken from TinkerForge and if we keep it that way, we can point to their documentation (with just adding documentation for the different way of triggering the stages).

https://github.com/Tinkerforge/esp32-firmware/blob/master/software/src/modules/evse_v2/preinit.cpp#L121-L157
https://github.com/Tinkerforge/warp-charger/blob/master/manual/manual_warp2.tex#L904-L914

**Comment by bs-github** [comment]
This fixes #1

File: software/src/modules/ac011k_hardware/ac011k_hardware.cpp [hunk 1]
```diff
@@ -116,22 +112,27 @@ void AC011KHardware::register_urls()
...
-    /* if (!btn && deadline_elapsed(last_btn_change + 10000)) { */
-    /*     logger.printfln("IO0 button was pressed for 10 seconds. Resetting to factory defaults."); */
-    /*     last_btn_change = millis(); */
-    /*     factory_reset_requested = true; */
-    /* } */
+    if (!btn && first && deadline_elapsed(last_btn_change + 10000)) {
+        logger.printfln("Button SW3 was pressed for 10 seconds. But resetting should be done on boot. Please use power off/on or SW1 to restart and press SW3 on boot to reset.");
+        first = false;
+        //logger.printfln("Button SW3 was pressed for 10 seconds. Resetting to factory defaults.");
+        //factory_reset_requested = true;
```
Comment (oelison) [inline software/src/modules/ac011k_hardware/ac011k_hardware.cpp:136]: simply esp reset.

**Review by oelison — CHANGES_REQUESTED** [review]
some comments added

**Pull request merged**